deepmerge/strategy/list.py

```diff
@@ -22,3 +22,7 @@ def strategy_prepend(config, path, base, nxt):
     def strategy_append(config, path, base, nxt):
         """ append nxt to base. """
         return base + nxt
+
+    @staticmethod
+    def strategy_append_unduplicated(config, path, base, nxt):
+        return base + list(set(nxt) - set(base))
```

Comment:
this approach may lead to some unspecified ordering.

Considering that lists are strictly ordered, thoughts on making base a set, then iterating through nxt and appending new elements as they're found?

I'm not particularly opinionated and I'm not sure others who use it would be either, but unspecified behavior can be quite problematic when dealing with unit tests that depend on consistency.

Reply:
I didn't know `set` orders items in a list :)
```
In [22]: base = [1, 3, 2]                                                                                                                                    
In [23]: nxt = [3, 5, 4, 1, 2]                                                                                                                               
In [24]: base + list(set(nxt) - set(base))                                                                                                                   
Out[24]: [1, 3, 2, 4, 5]
```

Reply:
sets don't order the item, but list -> set -> list will shuffle the ordering. In your case you're referring specifically to elements that are not present, so that behavior being less deterministic may not be an issue for many. But still good to address early on, this tends to get really difficult when unit testing multiple versions of python.